Patch-Source: https://github.com/ryanoasis/nerd-fonts/pull/860
--
From 37b9945956a6a367a4eb548ce07c0b5b509bc977 Mon Sep 17 00:00:00 2001
From: Fini Jastrow <ulf.fini.jastrow@desy.de>
Date: Thu, 7 Jul 2022 12:33:31 +0200
Subject: [PATCH] patcher: Fix grave (backtick) zero width

[why]
In ligature-enabled environments the accent grave will be rendered as
zero width, thus overlapping with the next character (in some source
fonts).

The problem is, that the glyph type in the sourcefonts is set to 'auto',
and fontforge exports these as 'mark' - the patched font will be broken.

[how]
There is no way we can get that glyph to be 'auto', but we can force it
to be an ordinary 'baseglyph' instead, and that will be respected on
export.

Maybe I should raise an Issue at fontforge... maybe later.

Fixes: #858
Fixes: #582

Signed-off-by: Fini Jastrow <ulf.fini.jastrow@desy.de>
---
 font-patcher | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/font-patcher b/font-patcher
index 7d063ab24..a665f472e 100755
--- a/font-patcher
+++ b/font-patcher
@@ -126,6 +126,14 @@ class font_patcher:
             symfont.close()
         print("\nDone with Patch Sets, generating font...")
 
+        # The grave accent and fontforge:
+        # If the type is 'auto' fontforge changes it to 'mark' on export.
+        # We can not prevent this. So set it to 'baseglyph' instead, as
+        # that resembles the most common expectations.
+        # This is not needed with fontforge March 2022 Release anymore.
+        if "grave" in self.sourceFont:
+            self.sourceFont["grave"].glyphclass="baseglyph"
+
         # the `PfEd-comments` flag is required for Fontforge to save '.comment' and '.fontlog'.
         if self.sourceFont.fullname != None:
             self.sourceFont.generate(self.args.outputdir + "/" + self.sourceFont.fullname + self.extension, flags=(str('opentype'), str('PfEd-comments')))
